lib/src/connector/connection.cc

```diff
@@ -498,6 +505,15 @@ WS_Context_Ptr Connection::onTlsInit(WS_Connection_Handle hdl)
                                   boost::asio::ssl::context::file_format::pem);
         ctx->load_verify_file(client_metadata_.ca);
 
+        // TODO validation and error handling:
+        // https://github.com/openssl/openssl/blob/e1dcac22607f6ecbb5ad4f15fc1a54f54f40c1ba/apps/lib/apps.c#L1202-L1234
+        // https://github.com/openssl/openssl/blob/e1dcac22607f6ecbb5ad4f15fc1a54f54f40c1ba/apps/lib/apps.c#L486-L518
+        BIO *crl_file = NULL;
```

Comment:
This is really the meat of this PR. I need to figure out all the validation we need to do here.